File: MVVMExampleLiveData/app/src/main/java/com/blackbirds/shakil/mvvmexamplelivedata/viewModel/MainActivityViewModel.java
package com.blackbirds.shakil.mvvmexamplelivedata.viewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel {

    MutableLiveData<String> resultLiveData;

    public MutableLiveData<String> addFunction(String valueOne, String valueTwo){
        int numOne = Integer.parseInt(valueOne);
        int numTwo = Integer.parseInt(valueTwo);
        int result = numOne + numTwo;

        if (resultLiveData == null){
            resultLiveData = new MutableLiveData<>();
        }
        resultLiveData.setValue(String.valueOf(result));

        return resultLiveData;
    }
}

File: KotlinMVVMExample/app/src/main/java/com/blackbirds/shakil/kotlinmvvmexample/ViewModel/MainActivityViewModel.kt
package com.blackbirds.shakil.kotlinmvvmexample.ViewModel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainActivityViewModel : ViewModel() {

    var resultMutableLiveData: MutableLiveData<String>? = null

    fun addFunction(valueOne: String, valueTwo: String): MutableLiveData<String> {
        val numOne = valueOne.toInt()
        val numTow = valueTwo.toInt()
        val result = numOne + numTow

        if (resultMutableLiveData == null){
            resultMutableLiveData = MutableLiveData()
        }
        resultMutableLiveData!!.value = result.toString()

        return resultMutableLiveData as MutableLiveData<String>
    }
}